src/server.c

```diff
@@ -3595,6 +3595,12 @@ void call(client *c, int flags) {
             updateCommandLatencyHistogram(&(real_cmd->latency_histogram), c->duration*1000);
     }
 
+    /* The duration needs to be reset after each call except for a blocked command,
+     * which has its duration accumulated on the client between each call. */
+    if (!(c->flags & CLIENT_BLOCKED && c->flags & CLIENT_PENDING_COMMAND)) {
```

Comment:
Why is it not enough to check just for `c->flags & CLIENT_BLOCKED`? Why do we need to also check for `CLIENT_PENDING_COMMAND`?

Reply:
I think it was placed to prevent wait command from crashing on the new assertion when it is timedout. 
@madolson I thkn that by making wait timeout to also call updateStatsOnUnblock should reduce the need for that